### ISSUE TITLE
[BUG] App Configuration Authorization Header uses comma instead of ampersand

### ISSUE THREAD

**Issue by mrm9084** [Azure/azure-sdk-for-java#5113]
Using a comma (,) is currently being used, but is not RFC compliant, (&) should be used instead.

**Comment by mssfang** [comment]
@mrm9084 Do you mean to change 
```java
        private static final String HMAC_SHA256 = "HMAC-SHA256 Credential=%s, SignedHeaders=%s, Signature=%s";
```
to 

```java
        private static final String HMAC_SHA256 = "HMAC-SHA256 Credential=%s& SignedHeaders=%s& Signature=%s";
```

**Comment by mrm9084** [comment]
Yes, but I also don't think the space should be after see: https://github.com/Azure/azure-sdk-for-net/blob/da5aaeafbc10ad318bcc291bc758fcf26c976db9/sdk/appconfiguration/Azure.Data.AppConfiguration/src/AuthenticationPolicy.cs#L73

**Comment by mssfang** [comment]
Thanks. Created a PR to fix it  Remove dot using ampersand #5549
